src/proof_rules/calculus/soundness_checker.rs

```diff
@@ -134,37 +323,43 @@ pub fn get_soundness_map(
     soundness_map
 }
 
-/// Track under-, over- and unknown approximating statements in a block
+/// Track under-, over- and unknown approximating statements in a block.
 ///
-/// Returns a list of spans and their corresponding approximation kinds of non-exact approximating statements
+/// Returns a list of spans and their corresponding approximation kinds of non-exact approximating statements.
 fn track_approximation_in_block(
     block: &Block,
     direction: Direction,
     calculus: Option<Calculus>,
     tcx: &mut TyCtx,
 ) -> ApproximationList {
-    block.node.iter().fold(Vec::new(), |mut acc, stmt| {
-        acc.extend(track_approximation_in_statement(
-            stmt, direction, calculus, tcx,
-        ));
-        acc
-    })
+    block
```

Comment:
I think the cleaner version of this would be to do a `map` first, and then `flatten` and then collect the result into the ApproximationList, no?

At this point, since ApproximationList doesn't seem to carry any guarantees, a no-newtype simple Vec type alias for ApproximationList would make this easier.